chapters/system_architecture/sections/chunk_worker/subsections/saving.tex

```diff
@@ -0,0 +1,6 @@
+\subsection{Saving chunks}
+Saving chunks is handled by a process similar to the one discussed in \autoref{subsec:loading-and-generating}.
+On each frame, the main thread in the \texttt{DeleteChunks} function determines which chunks are too far from the player and thus should be removed from the game and saved to disk.
+Positions of these chunks are enqueued into the \texttt{chunksToSave} queue, and their resources are freed.
+
+The worker thread dequeues chunks' positions from the \texttt{chunksToSave} queue and saves the scalar field associated with a given chunk on the disk.
```

Comment:
Might be worth it to add that there is a system that makes sure that chunks are not generated/deleted constantly if the player moves between chunk borders. Only the "oldest" chunks are deleted. SOme that are further away than RednerDistance stay loaded for some time

Reply:
Take a look if it's ok